scipy-stubs/signal/_spectral_py.pyi

```diff
@@ -1,97 +1,206 @@
-from typing import Literal
+from collections.abc import Callable
+from typing import Literal, TypeAlias, overload
+from typing_extensions import Unpack
 
-from scipy._typing import Untyped, UntypedCallable
+import numpy as np
+import numpy.typing as npt
+import optype as op
+from numpy._typing import _ArrayLikeFloat_co, _ArrayLikeNumber_co
+from scipy._typing import AnyInt, AnyReal
+from .windows._windows import _Window, _WindowNeedsParams
 
 __all__ = ["check_COLA", "check_NOLA", "coherence", "csd", "istft", "lombscargle", "periodogram", "spectrogram", "stft", "welch"]
 
-def lombscargle(x: Untyped, y: Untyped, freqs: Untyped, precenter: bool = False, normalize: bool = False) -> Untyped: ...
+_Array_f8: TypeAlias = npt.NDArray[np.float64]
+_Array_f8_1d: TypeAlias = np.ndarray[tuple[int], np.dtype[np.float64]]
+_ArrayFloat: TypeAlias = npt.NDArray[np.float32 | np.float64 | np.float128]
+_ArrayComplex: TypeAlias = npt.NDArray[np.complex64 | np.complex128 | np.complex256]
+
+_GetWindowArgument: TypeAlias = _Window | tuple[_Window | _WindowNeedsParams, Unpack[tuple[object, ...]]]
+_Detrend: TypeAlias = Literal["literal", "constant", False] | Callable[[npt.NDArray[np.generic]], npt.NDArray[np.generic]]
+_Scaling: TypeAlias = Literal["density", "spectrum"]
+_LegacyScaling: TypeAlias = Literal["psd", "spectrum"]
```

Comment:
This is because of the documentation for `stft` and `istft`. See https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.stft.html

The `scaling` parameter type differs from the non-legacy functions so I decided to call this type `_LegacyScaling` to reflect that this is scaling argument from a legacy function.

Reply:
> This is because of the documentation for `stft` and `istft` says that these functions are legacy. See https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.stft.html
> 
> The `scaling` parameter type differs from the non-legacy functions so I decided to call this type `_LegacyScaling` to reflect that this is scaling argument from a legacy function.